lib/presentation/molecules/youtube_player_molecule.dart

```diff
@@ -0,0 +1,47 @@
+import 'package:flutter/material.dart';
+import 'package:youtube_player_flutter/youtube_player_flutter.dart';
+
+class YoutubePlayerMolecule extends StatelessWidget {
+  const YoutubePlayerMolecule({
+    required this.url,
+    super.key,
+    this.hideThumbnail = false,
+    this.showVideoProgressIndicator = true,
+    this.progressIndicatorColor = Colors.red,
+    this.playedColor = Colors.red,
+    this.handleColor = Colors.red,
+    this.padding = const EdgeInsets.all(8.0),
+  });
+  final String url;
+  final bool hideThumbnail;
+  final bool showVideoProgressIndicator;
+  final Color progressIndicatorColor;
+  final Color playedColor;
+  final Color handleColor;
+  final EdgeInsetsGeometry padding;
+
+  @override
+  Widget build(BuildContext context) {
+    final YoutubePlayerController controller = YoutubePlayerController(
+      initialVideoId: YoutubePlayer.convertUrlToId(url) ?? "",
```

Comment:
What will it show if initialVideoId getting ""

Reply:
I have added ! operator instead of ?? because when we reach this part we will have a valid video ID anyway